python/ray/tests/test_gpu_objects_gloo.py

```diff
@@ -24,10 +25,60 @@ def get_gpu_object(self, obj_id: str):
         )
         if gpu_object_store.has_gpu_object(obj_id):
             gpu_object = gpu_object_store.get_gpu_object(obj_id)
-            print(f"gpu_object: {gpu_object}")
             return gpu_object
         return None
 
+    def get_num_gpu_objects(self):
+        gpu_object_manager = ray._private.worker.global_worker.gpu_object_manager
+        return len(gpu_object_manager.gpu_object_store.gpu_object_store)
+
+
+@pytest.mark.parametrize("data_size_bytes", [100])
+def test_gc_gpu_object(ray_start_regular, data_size_bytes):
+    """
+    This test covers different code paths for reference counting
+    by controlling the size of the CPU data to be inlined or not.
+
+    For small data, GPU objects are inlined, but the actual data lives
+    on the remote actor. Therefore, if we decrement the reference count
+    upon inlining, we may cause the tensors on the sender actor to be
+    freed before transferring to the receiver actor.
```

Comment:
I think this comment is not really relevant right now. Better to just say what the test does.

Reply:
updated